### ISSUE TITLE
Return a deleted diff for a deleted file in the compare API

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-server#11364]
### Describe the problem and steps to reproduce it:

- Sign in to the code manager
- Load a compare view for a version where a file has been deleted, such as https://code.addons-dev.allizom.org/en-US/compare/616645/versions/1655530...1655531/?path=scripts%2Fcontent.js
- Click on `scripts/background.js`, which has been deleted in this version

### What happened?

A 500 error

https://sentry.prod.mozaws.net/operations/olympia-dev/issues/5547253/?query=is:unresolved

```
KeyError: 'scripts/background.js'
  File "rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "olympia/reviewers/views.py", line 1403, in retrieve
    return Response(serializer.data)
  File "rest_framework/serializers.py", line 563, in data
    ret = super(Serializer, self).data
  File "rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "olympia/addons/serializers.py", line 177, in to_representation
    return super(SimpleVersionSerializer, self).to_representation(instance)
  File "rest_framework/serializers.py", line 530, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "rest_framework/serializers.py", line 530, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "rest_framework/fields.py", line 1889, in to_representation
    return method(value)
  File "olympia/reviewers/serializers.py", line 206, in get_download_url
    blob_or_tree = tree[selected_file]
```

### What did you expect to happen?

This should return a deleted diff so that the code manager can display all previous lines of the code. This will help the reviewer understand what was deleted.

### Anything else we should know?

There was a recent fix in https://github.com/mozilla/addons-server/pull/11346 for non-existent files but this is a slightly different case.

**Comment by willdurand** [comment]
> There was a recent fix in #11346 for non-existent files but this is a slightly different case.

Interesting, so it looks like the requested file is part of the files but not part of the `blob_or_tree` (git data).

**Comment by EnTeQuAk** [comment]
> Interesting, so it looks like the requested file is part of the files but not part of the blob_or_tree (git data).

Correct, because we explicitly add deleted files to the list of files but mark them accordingly.

**Comment by AlexandraMoga** [comment]
Verified fixed on -dev.  I've tested with several examples and the 500 error can't be reproduced anymore:

API:
https://addons-dev.allizom.org/api/v5/reviewers/addon/616548/versions/1655557/compare_to/1655558/?file=_locales/it/messages.json


Code manager:
https://code.addons-dev.allizom.org/en-US/compare/616645/versions/1655530...1655531/?path=scripts%2Fbackground.js
https://code.addons-dev.allizom.org/en-US/compare/616847/versions/1688177...1688178/?path=coinminer.js
